torch/csrc/inductor/aot_runtime/model.h

```diff
@@ -330,30 +351,26 @@ class AOTInductorModel : public AOTInductorModelBase<AOTInductorModel> {
   }
 };
 
-#define AOTI_TORCH_ERROR_CODE_CHECK(call)                                 \
-  if ((call) != AOTI_TORCH_SUCCESS) {                                     \
-    throw std::runtime_error(                                             \
-        std::string(#call " API call failed at ") + __FILE__ + ", line" + \
-        std::to_string(__LINE__));                                        \
-  }
-
-using RAIIAtenTensorHandle = std::shared_ptr<AtenTensorOpaque>;
-
-inline RAIIAtenTensorHandle create_raii_tensor_handle_for_extern(
-    AtenTensorHandle handle) {
-  return RAIIAtenTensorHandle(handle, [](AtenTensorHandle ptr) {
-    // Do nothing for extern tensor handles
-  });
-}
-
-inline RAIIAtenTensorHandle create_raii_tensor_handle_for_temp(
+// TODO: will update RAIIAtenTensorHandle to unique_ptr and change this
+inline RAIIAtenTensorHandle steal_tensor_handle_from_raw_to_raii(
     AtenTensorHandle handle) {
   return RAIIAtenTensorHandle(handle, [](AtenTensorHandle ptr) {
     AOTI_TORCH_ERROR_CODE_CHECK(
         aoti_torch_delete_tensor_object(static_cast<AtenTensorHandle>(ptr)));
   });
 }
 
+// TODO: will update RAIIAtenTensorHandle to unique_ptr and change this
+std::vector<AtenTensorHandle> raii_handles_to_raw_handles(
```

Comment:
It is helpful to know that, although this allocates a vector, it only is doing borrows on all the handles.

The repeated O(n) copies to go from RAII to not RAII is quite tiresome, especially because nothing "actually" happens here. You can avoid having to do the copy if you make RAII class for the entire vector, and then just make sure you store the contents of the vector in a canonical form, though I'm not exactly sure if you should go straight there yet.